src/main/kotlin/com/ebay/plugins/metrics/develocity/GatherHourlyTask.kt

```diff
@@ -179,13 +182,20 @@ open class GatherHourlyTask @Inject constructor(
             val detailsQuery = BuildQuery(
                 models = modelsNeeded
             )
-            channel.consumeEach { buildRef ->
-                develocityServiceProperty.get().build(buildRef.id, detailsQuery) { build ->
+            for (buildRef in channel) {
+                processingBuilds.add(buildRef.id)
+                val build = withTimeoutOrNull(2.minutes) {
```

Comment:
It will be nice to expose the timeout as a configuration for the plugin, some teams might want to tweak, for now Is fine as it is.

Reply:
Good call.